src/app/public/modules/split-view/split-view.component.ts

```diff
@@ -0,0 +1,375 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+
+import {
+  animate,
+  style,
+  transition,
+  trigger
+} from '@angular/animations';
+
+import {
+  SkyAppWindowRef,
+  SkyCoreAdapterService,
+  SkyMediaBreakpoints,
+  SkyMediaQueryService
+} from '@skyux/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  Subscription
+} from 'rxjs/Subscription';
+
+import 'rxjs/add/operator/take';
+
+import 'rxjs/add/operator/takeUntil';
+
+import 'rxjs/add/operator/takeWhile';
+
+import 'rxjs/add/observable/fromEvent';
+
+import {
+  SkySplitViewBeforeWorkspaceCloseHandler
+} from './types/split-view-before-workspace-close-handler';
+
+import {
+  SkySplitViewMessage
+} from './types/split-view-message';
+
+import {
+  SkySplitViewMessageType
+} from './types/split-view-message-type';
+
+import {
+  SkySplitViewMediaQueryService
+} from './split-view-media-query.service';
+
+import {
+  SkySplitViewDrawerComponent
+} from './split-view-drawer.component';
+
+import {
+  SkySplitViewWorkspaceComponent
+} from './split-view-workspace.component';
+
+let nextId = 0;
+
+@Component({
+  selector: 'sky-split-view',
+  templateUrl: './split-view.component.html',
+  styleUrls: ['./split-view.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger(
+      'blockAnimationOnLoad', [
+        transition(':enter', [])
+      ]
+    ),
+    trigger(
+      'drawerEnter', [
+        transition(`void => *`, [
+          style({transform: 'translate(-100%)'}),
+          animate('150ms ease-in')
+        ])
+      ]
+    ),
+    trigger(
+      'workspaceEnter', [
+        transition(`void => *`, [
+          style({transform: 'translate(100%)'}),
+          animate('150ms ease-in')
+        ])
+      ]
+    )
+  ]
+})
+export class SkySplitViewComponent implements OnInit, AfterViewInit, OnDestroy {
+
+  @Input()
+  public backLabel: string;
```

Comment:
What do you think about naming this `backButtonText`, to match our wording on other components' inputs? The `@skyux/modals` library is a good example; it uses "text" to describe the content of a button element in many places. See: https://github.com/blackbaud/skyux-modals/blob/master/src/app/public/modules/confirm/types/confirm-button-config.ts#L3

Reply:
Done.